src/daemon/client.c

```diff
@@ -273,6 +277,82 @@ client_handle_get_interfaces(struct lldpd *cfg, enum hmsg_type *type,
 	return output_len;
 }
 
+#ifdef ENABLE_LLDPMED
+/**
+ * Set local chassis inventory info
+ * Input: chassis object
+ * Output: updated chassis object
+ */
+static ssize_t
+client_handle_set_local_chassis(struct lldpd *cfg, enum hmsg_type *type,
+    void *input, int input_len, void **output, int *subscribed)
+{
+	struct lldpd_chassis *chassis = NULL;
+	struct lldpd_chassis *local_chassis = NULL;
+	struct utsname un;
+
+	log_debug("rpc", "client request a change in chassis configuration");
+	if (lldpd_chassis_unserialize(input, input_len, &chassis) <= 0) {
+		*type = NONE;
+		return 0;
+	}
+
+	local_chassis = LOCAL_CHASSIS(cfg);
+
+	free(local_chassis->c_med_hw);
+	local_chassis->c_med_hw = (!chassis->c_med_hw) ? dmi_hw() : strdup(chassis->c_med_hw);
+
+	// Follows lldpd.c - only set sw if advertising is enabled
+	if (cfg->g_config.c_advertise_version) {
+		free(local_chassis->c_med_sw);
+
+		if (!chassis->c_med_sw) {
+			if (uname(&un) < 0) {
+				log_warn("rpc", "Could not get default uname. Will continue anyway.");
+				local_chassis->c_med_sw = NULL;
+			}
+			else
```

Comment:
Put else on the same line as the closing brace.

Reply:
Fixed in latest update.